source/extensions/filters/http/grpc_streaming/config.cc

```diff
@@ -0,0 +1,133 @@
+#include "extensions/filters/http/grpc_streaming/config.h"
+
+#include "envoy/http/filter.h"
+#include "envoy/registry/registry.h"
+
+#include "common/grpc/common.h"
+#include "common/grpc/context_impl.h"
+
+#include "extensions/filters/http/grpc_streaming/message_counter.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace HttpFilters {
+namespace GrpcStreaming {
+
+namespace {
+
+class GrpcStreamingFilter : public Http::StreamFilter {
+public:
+  explicit GrpcStreamingFilter(Grpc::Context& context) : context_(context) {}
+
+  // Http::StreamFilterBase
+  void onDestroy() override {}
+
+  // Http::StreamDecoderFilter
+  Http::FilterHeadersStatus decodeHeaders(Http::HeaderMap& headers, bool) override {
+    grpc_request_ = Grpc::Common::hasGrpcContentType(headers);
+    if (grpc_request_ && decoder_callbacks_) {
+      cluster_ = decoder_callbacks_->clusterInfo();
+      if (cluster_) {
+        request_names_ = context_.resolveServiceAndMethod(headers.Path());
+      }
+    }
+    return Http::FilterHeadersStatus::Continue;
+  }
+  Http::FilterDataStatus decodeData(Buffer::Instance& data, bool) override {
+    if (grpc_request_) {
+      uint64_t delta = IncrementMessageCounter(data, &request_counter_);
```

Comment:
drive by: Should IncrementMessageCounter be a method of Counter?
BTW: lower case `increaseMessageCounter`